matrix-sdk-android/src/main/java/fr/gouv/tchap/android/sdk/internal/session/room/TchapRoomGetter.kt

```diff
@@ -59,14 +59,24 @@ internal class TchapRoomGetter @Inject constructor(
         // The case left-x isn't possible because we ignore for the moment the left rooms.
         // If other member user id is an email, we take the oldest room.
 
-        return directRoomMemberships.firstOrNull { it.first == Membership.JOIN && it.second == Membership.JOIN }?.roomId // join - join
-                ?: directRoomMemberships.firstOrNull { it.first == Membership.INVITE && it.second == Membership.JOIN }?.roomId // invite - join
-                ?: directRoomMemberships.firstOrNull { it.first == Membership.JOIN && it.second == Membership.INVITE }?.roomId // join - invite
-                ?: directRoomMemberships.firstOrNull { it.first?.isActive() == true && it.second == Membership.LEAVE }?.roomId // join or invite - left
-                ?: directRoomMemberships // otherUserId is an email
-                        .takeIf { Patterns.EMAIL_ADDRESS.matcher(otherUserId).matches() }
-                        ?.firstOrNull { it.first == Membership.JOIN }
-                        ?.roomId
+        return when {
+            directRoomMemberships.firstOrNull { it.first == Membership.JOIN && it.second == Membership.JOIN }?.roomId != null ->
+                directRoomMemberships.firstOrNull { it.first == Membership.JOIN && it.second == Membership.JOIN }?.roomId // join - join
+            directRoomMemberships.firstOrNull { it.first == Membership.INVITE && it.second == Membership.JOIN }?.roomId != null ->
+                directRoomMemberships.firstOrNull { it.first == Membership.INVITE && it.second == Membership.JOIN }?.roomId // invite - join
+            directRoomMemberships.firstOrNull { it.first == Membership.JOIN && it.second == Membership.INVITE }?.roomId != null ->
+                directRoomMemberships.firstOrNull { it.first == Membership.JOIN && it.second == Membership.INVITE }?.roomId // join - invite
+            directRoomMemberships.firstOrNull { it.first?.isActive() == true && it.second == Membership.LEAVE }?.roomId != null ->
+                null // join or invite - left. We want to create a new room if the other left the room.
+            directRoomMemberships // otherUserId is an email
+                            .takeIf { Patterns.EMAIL_ADDRESS.matcher(otherUserId).matches() }
+                            ?.firstOrNull { it.first == Membership.JOIN }
+                            ?.roomId != null -> directRoomMemberships // otherUserId is an email
+                    .takeIf { Patterns.EMAIL_ADDRESS.matcher(otherUserId).matches() }
+                    ?.firstOrNull { it.first == Membership.JOIN }
+                    ?.roomId
+            else -> null
+        }
```

Comment:
I prefer the previous format which was easier to read IMO and has the advantage of not duplicating the operation.